### DISCUSSION TITLE
Possible to have multiple QueryClients and tell specific query which to use?

### DISCUSSION THREAD

**Discussion by hmbrg** [TanStack/query#1877] [Q&A]
I'm building a library that uses `react-query`. 
I want to provide users the option to set a default query function via a wrapper around the `QueryClientProvider`. 

How do I make sure I don't overwrite any other instances of `react-query` the user might be using in his project?

Thanks in advance!

**Comment by TkDodo** [comment]
the Providers are just react-context, so `useQueryClient` will just lookup the nearest Provider. You can make your api in a way that an optional Provider is passed, and you can maybe work with `queryClient.setQueryDefaults` ...

**Reply by hmbrg** [reply]
How do I make sure that `useQuery` is not using any other `QueryClientProvider`s ? I can't seem to find any options for that.

**Reply by TkDodo** [reply]
it will use the _nearest_ `QueryClientProvider` in the react component tree. That's how react context works.
